doc-src/ch-test_desc.rst

```diff
@@ -3,159 +3,178 @@ Synopsis
-    Test the example application images.
+* *Unpacked images directory*. Images are unpacked into and then run from
+  here.
 
-    This phase requires contents in image directory (:code:`--img-dir`) produced
-    by a successful :code:`run` phase execution.
+* *Filesystem permissions* director[y|ies]. These are used to test that the
+  kernel is enforcing permissions correctly. Note that this exercises the
+  kernel, not Charliecloud, and can be omitted from routine Charliecloud
+  testing.
 
-    These tests are run using a slurm allocation if one is available or single
-    node (localhost) if not.
+The first three are created when needed if they don't exist, while the
+filesystem permissions fixtures must be created manually, in order to
+accommodate configurations where sudo is not available via the same login path
+used for running tests.
 
-  :code:`mk-perm-dirs`
-    Create a series of files and test fixtures used to test file
-    permission enforcement.
+Some of the tests exercise parallel functionality. If :code:`ch-test` is run
+on a single node, multiple cores will be used; if in a Slurm allocation,
+multiple nodes.
 
-    Requires :code:`--sudo` and :code:`--perm-dirs DIR` arguments
+The subset of tests to run mostly splits along two key dimensions. The *phase*
+is which parts of the workflow to run. Different parts of the workflow can be
+tested on different systems by copying the necessary artifacts between them,
+e.g. by building images on one system and running them on another. The *scope*
+allows trading off thoroughness versus time.
 
-  :code:`rm-perm-dirs`
-    Remove test fixtures for kernel file permission enforcement.
+:code:`PHASE`, if specified, must be one of:
 
-    Requires :code:`--sudo` and :code:`--perm-dirs DIR` arguments.
-
-  :code:`clean`
-    Delete test artifacts in :code:`--img-dir DIR` and :code:`--tar-dir DIR`.
+  :code:`build`
+    Image building and associated functionality, with the selected builder.
 
-.. note::
-    If no phase is specified, :code:`ch-test` will execute the following
-    phases in order: :code:`build`, :code:`run`, and :code:`examples`
+  :code:`run`
+    Running containers and associated functionality. This requires a packed
+    images directory produced by a successful :code:`build` phase, which can
+    be copied from the build system if it's not also the run system.
 
-Optional arguments:
+  :code:`examples`
+    Example applications. Requires an unpacked images directory produced by a
+    successful :code:`run` phase.
 
-  :code:`-b`, :code:`--builder`
-    Specify prefered image builder. Default: :code:`$CH_BUILDER`.
+  :code:`mk-perm-dirs`
+    Create the filesystem permissions directories. Requires
+    :code:`--perm-dirs`.
 
-  :code:`--dry-run`
-    Print test suite phase details without executing.
+  :code:`clean`
+    Delete test artifacts on disk and all images in builder storage (whether
+    or not Charliecloud-related), except for the permissions test fixtures.
```

Comment:
Spicy. Not sure I'd like `clean` to wipe my docker/buildah image cache. Am I misunderstanding this?

Reply:
Your understanding is correct. Why don't we leave it alone for now (i.e., don't wipe builder storage). I would like an easy way to do that though; our current `test/docker-clean.sh` isn't very user-facing.